### DISCUSSION TITLE
How to do intersphinx reference to confval in Sphinx docs?

### DISCUSSION THREAD

**Discussion by jdillard** [sphinx-doc/sphinx#12151] [Q&A]
What is the recommended way to reference a `confval` in the Sphinx docs? It seems it would be:
```
:confval:`sphinx:extensions`
```
but that results in:
```
ERROR: Unknown interpreted text role "confval".
```
I can add the following to `conf.py` to fix:
```
def setup(app):
    app.add_object_type('confval', 'confval',
                        objname='Sphinx configuration value',
                        indextemplate='pair: %s; Sphinx configuration value')
```
but that seems overly complex.

Am I doing it wrong? Should the `confval` object type be nativity available for cross reference purposes?

**Comment by chrisjsewell** [comment]
oii @jdillard wait your turn, I was literally just typing about `confval` in #12152  😆

**Reply by jdillard** [reply]
haha what are the odds! I can close this and add it as part of the discussion there

**Reply by chrisjsewell** [reply]
I know right, yeh exactly was just finishing it up the final section and then definitely open to feedback 👌 (done)

**Comment by jdillard** [comment]
Discussion moved to https://github.com/orgs/sphinx-doc/discussions/12152
